internal/app/cloudwatcher/aggregation.go

```diff
@@ -0,0 +1,109 @@
+package cloudwatcher
+
+import (
+	"gonum.org/v1/gonum/floats" // Float math tools.
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+)
+
+// AggregationData represents a container for some data series
+// about Elasticsearch nodes, that can be converted
+// into a CloudWatch Metrics data point.
+type AggregationData interface {
+	// AddSample adds the given node to the data series
+	AddSample(*NodeStats)
+
+	// Datum returns the aggregated data as a CloudWatch
+	// Metrics data point.
+	Datum() cloudwatch.MetricDatum
+}
+
+// SelectorFn returns some aspect of a NodeStats to
+// be aggregated. If the node shouldn't count towards
+// a data set, the selector function should return nil.
+type SelectorFn func(*NodeStats) *float64
+
+// StatsData aggregates count, min, max, and sum of the samples.
+type StatsData struct {
+	Template cloudwatch.MetricDatum
+	Selector SelectorFn
+
+	data []float64
+}
+
+var _ AggregationData = (*StatsData)(nil)
```

Comment:
I'm pretty confused about the syntax and overall purpose of this line.  What is it doing exactly?

Reply:
And if this is some kind of subclassing thing, is there a way to only define AddSample once on AggregationData so that we don't need four of the same method?

Reply:
This is asserting that the `StatsData` type implements the `AggregationData` interface by casting a nil `StatsData` as a `AggregationData`. If `StatsData` didn't implement the interface, this would throw a compiler error. The single underscore variable name (`_`) is called a [blank identifier](https://stackoverflow.com/questions/24357028/meaning-of-underscore-blank-identifier-in-go), which gets optimized out by the compiler.  I'll add a comment to explain.